src/main/scala/com/databricks/labs/overwatch/validation/SnapValidation.scala

```diff
@@ -179,15 +217,27 @@ class SnapValidation(params: SnapValidationParams,
    */
   def executeSnapshots(): Dataset[SnapReport] = {
     // state tables clone and reset for silver and gold modules
-    snapStateTables()
-    resetPipelineReportState(params.snapDatabaseName)
+    case class ModuleTarget(module: Module, target: PipelineTable)
 
-    // snapshots bronze tables, returns ds with report
-    val bronzeTargets = (Bronze(workspace).getAllTargets).par
-    bronzeTargets.tasksupport = taskSupport
+    val bronzePipeline = Bronze(workspace)
+    val bronzeTargets = bronzePipeline.BronzeTargets
+    snapStateTables(bronzePipeline)
+    resetPipelineReportState(bronzePipeline.pipelineStateTarget)
 
-    bronzeTargets
-      .map(bronzeTarget => snapTable(bronzeTarget))
+    // snapshots bronze tables, returns ds with report
+    val bronzeTargetsWModule = bronzePipeline.config.overwatchScope.map {
```

Comment:
Only snap relevant Scopes. Allows user to minimize test breadth and avoid errors if some scopes are disabled.